### DISCUSSION TITLE
Loadstructure  e LoadFromJson

### DISCUSSION THREAD

**Discussion by ds-sampaio** [viniciussanchez/dataset-serialize#172] [Q&A] [answered]
Caros colegas, estou tentando usar o Loadstructure  juntamente com LoadFromJson, pois preciso que minha Memtable tenha os tipos corretos de cada campo.
meu código é esse

FDMemTable.LoadStructure(LResponse.JSONValue.GetValue<TJSONArray>('structure'),False);
FDMemTable.LoadFromJSON(LResponse.JSONValue.GetValue<TJSONObject>('data'), False)

e minha estrutura é essa:

{ "structure": [
        {
            "fieldName": "id_historico",
            "dataType": "ftInteger",
            "size": 0
        },
        {
            "fieldName": "descricao",
            "dataType": "ftString",
            "size": 500
        },
        {
            "fieldName": "data",
            "dataType": "ftDate",
            "size": 0
        },
        {
            "fieldName": "hora",
            "dataType": "ftTime",
            "size": 0
        },
        {
            "fieldName": "codigo",
            "dataType": "ftInteger",
            "size": 0
        },
        {
            "fieldName": "historico_ref",
            "dataType": "ftInteger",
            "size": 0
        },
        {
            "fieldName": "id_empresa",
            "dataType": "ftInteger",
            "size": 0
        },
        {
            "fieldName": "cod_ref",
            "dataType": "ftInteger",
            "size": 0
        }
    ]
}

porém os dados não estao aparecendo no meu dbgrid. Sendo que antes de utilizar o LoadStrtucture estava carregando normalmente o dbgrid

**Comment by viniciussanchez** [comment, marked as answer]
Como está do data do seu json?

**Reply by ds-sampaio** [reply]
"data": {
        "id_historico": 1,
        "descricao": "Teste de historico fbfbjkf",
        "data": "1995-08-03",
        "hora": "00:00:00",
        "codigo": 57,
        "historico_ref": 11,
        "id_empresa": 1,
        "cod_ref": 12
    }

**Reply by viniciussanchez** [reply]
São projetos diferentes que você está fazendo isso? Por exemplo, um projeto está gerando esse JSON e outro está consumindo? Porque se for tem que fazer a configuração do nome dos fields, por exemplo tudo minúsculo, tudo maiúsculo, ou CamelCase, lowerCamelCase. Ambos os projetos tem que estar iguais

**Reply by ds-sampaio** [reply]
Tipo, quem gera esse json é o meu backend ai o front pega esse json completo e refatia entre o data e o structure
seria algo desse tipo: 
{data : 
 {objeto},
structure: 
 [{ objetos} }]

lá no backend é tudo minusculo.
no front eu usei o TDataSetSerializeConfig.GetInstance.CaseNameDefinition := TCaseNameDefinition.cndLower;

**Reply by ds-sampaio** [reply]
Eu descobrir o que é, no DataSet Serialize quando dou um SaveStructure ele tem mais campos do que o meu Json Structure gerado no backend, fui tirando campo a campo para testar. E tem um campo que é o Visible, assim q removi ele não apareceu as informações no grid.

No dataset serialize o formato vem desse jeito

[
  { "alignment":"taRightJustify",
    "fieldName":"ID_HISTORICO",
	"displayLabel":"ID_HISTORICO",
	"dataType":"ftInteger",
	"size":0,
	"origin":"ID_HISTORICO",
	"key":true,
	"required":true,
	"visible":true,
	"readOnly":false,
	"autoGenerateValue":"arNone"
   }
]

ja no meu so vem

{
"fieldName": "id_historico",
"dataType": "ftInteger",
"size": 0
}
no meu json teria q vir o campo Visible : True para funcionar
problema resolvido. Terei q mudar meu backend
Obrigada Vinicius

**Reply by viniciussanchez** [reply]
Legal, que bom que descobriu...

**Reply by ds-sampaio** [reply]
Para quem tiver o mesmo problema deixaria aq documentado que o visible em questão fica, na unit DataSet.Serialize.Utils
no metodo  NewDataSetField na linha 240 Result.Visible := AFieldStructure.Visible;
Estarei fechando aq cmo resolvido. Mais uma vez Obrigada Vinicius
